packages/template-retail-react-app/app/components/cancel-order-modal/index.jsx

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import React from 'react'
+import PropTypes from 'prop-types'
+import {FormattedMessage} from 'react-intl'
+import {
+    Modal,
+    ModalOverlay,
+    ModalContent,
+    ModalHeader,
+    ModalFooter,
+    ModalBody,
+    ModalCloseButton,
+    Button,
+    Text
+} from '@chakra-ui/react'
+
+/**
+ * A Modal for requesting order cancellation
+ */
+const CancelOrderModal = ({
+    isOpen,
+    onClose,
+    order = null,
+    onRequestCancellation = null,
```

Comment:
```suggestion
    order,
    onRequestCancellation,
```

There's a difference between`null` vs  `undefined`. I think of `null` as something intentionally passed in. 

But in this case, I don't think it's necessary to add null or undefined. Your component propTypes already specifies which props are required and optional.

Reply:
I've also seen noops set as default for some modals, not sure what difference that makes because the values always seem to be populated (required)